pkg/block/s3/retyer.go

```diff
@@ -0,0 +1,41 @@
+package s3
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+// errConnectionResetRetryer implements the `request.Retryer` interface and allows for
+// customization of the retry behaviour of an AWS client.
+// errConnectionResetRetryer was taken from cockroachdb
+// https://github.com/cockroachdb/cockroach/blob/7a8cac9b11036dcf401453eb36e531fe381b70b4/pkg/cloud/amazon/s3_storage.go#L95
+type errConnectionResetRetryer struct {
+	client.DefaultRetryer
+}
+
+// isErrReadConnectionReset returns true if the underlying error is a read
+// connection reset error.
+//
+// NB: A read connection reset error is thrown when the SDK is unable to read
+// the response of an underlying API request due to a connection reset. The
+// DefaultRetryer in the AWS SDK does not treat this error as a retryable error
+// since the SDK does not have knowledge about the idempotence of the request,
+// and whether it is safe to retry -
+// https://github.com/aws/aws-sdk-go/pull/2926#issuecomment-553637658.
+//
+// In lakeFS all operations with s3 (read, write, list) are considered idempotent,
```

Comment:
Would it make sense to ensure that only one of these operations is being performed?  Otherwise someone might _add_ a new operation and not think to avoid it being retried.

Reply:
That's not an exhaustive list, other operations may also benefit from it. I'll update the comment to reflect that.

Notice that even before this change, the caller must always assumes that a retry may happen. This change only adds one more case where a specific tcp error is retried.